### DISCUSSION TITLE
How to specify OnReject delegate handler for slide window rate limit strategy? Also can I combine mulitple rate limit strategy for a single pipeline?

### DISCUSSION THREAD

**Discussion by AsuraKev** [App-vNext/Polly#1945] [Q&A] [answered]
In the doc, there are multiple variation of stragtegies in rate limiting however only the concurrent rate limit can allow specifying OnReject handler. I am wondering how to achieve the same with Slide window strategy?
![image](https://github.com/App-vNext/Polly/assets/52765572/df24417a-badb-4e2c-80fd-d7979f0cbbb7)

Another question is whether we can combine mulitple rate limit strategy on a single pipeline?

thanks! :)

**Comment by martincostello** [comment, marked as answer]
> Another question is whether we can combine multiple rate limit strategy on a single pipeline?

You can [chain rate limiters together](https://learn.microsoft.com/aspnet/core/performance/rate-limit#create-chained-limiters), so you could create a custom limiter made of many others, and then use the `RateLimiter` delegate to provide a lease from the chained rate limiter.

**Reply by martintmk** [reply]
I might add that Polly gives great flexibility on how you define the rate limiter. We made simple scenarios easy, but otherwise we do not put any restrictions around how you use rate limiters. 

For example:
https://www.pollydocs.org/strategies/rate-limiter.html#partitioned-rate-limiter

**Reply by AsuraKev** [reply]
Thanks guys for the help. Regarding the sliding window strategy, how can it work together with a retry strategy when the exception contains **RetryAfter** information? In the doc, it says to implement the **OnRejected** callback so that **RateLimiterRejectedException** will not be thrown. Since **SlidingWindowRateLimiterOptions** doesnt have allow us to specify this callback, how can this be achieved? 
:)
